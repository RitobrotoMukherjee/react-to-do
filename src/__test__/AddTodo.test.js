import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import InputTodo from '../components/AddTodo';

const mockTodoParent = () => 'Hello';

describe('Input and Header test', () => {
  it('Input todo component testing', () => {
    const TREE = renderer.create(<InputTodo addTodo={mockTodoParent} />).toJSON();
    expect(TREE).toMatchSnapshot();
  });

  it('Test Input Todo', async () => {
    // Arrange
    render(<InputTodo addTodo={mockTodoParent} />);
    const input = screen.getByRole('textbox');

    // ACT
    fireEvent.change(input, {
      target: { value: 'Workout' },
    });

    // Assert
    expect(input.value).toBe('Workout');
  });
});
