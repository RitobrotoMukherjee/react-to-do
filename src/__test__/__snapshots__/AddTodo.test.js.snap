// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input and Header test Input todo component testing 1`] = `
<form
  className="Add-to-do-form"
  onSubmit={[Function]}
>
  <input
    className="Form-control Add-to-do-input"
    name="title"
    onChange={[Function]}
    placeholder="Add A To Do"
    value=""
  />
</form>
`;
